new-website/utils/models/get_models.py

```diff
@@ -0,0 +1,143 @@
+"""
+Python script to scrape the all the models information from the official deepchem documentation
+"""
+
+import os
+import json
+import dataclasses
+from json import JSONEncoder
+from bs4 import BeautifulSoup
+import requests
+
+DOC_URL = 'https://deepchem.readthedocs.io/en/latest/api_reference/models.html#model-cheatsheet'
+REDIRECT_URL = 'https://deepchem.readthedocs.io/en/latest/api_reference/models.html#'
+TABLES = {
+    'id81': 'General',
+    'id82': 'Molecule',
+    'id83': 'Material'
+}
+model_list = []
+backend_list = []
+type_list = []
+featurizer_list = []
+
+@dataclasses.dataclass
+class Model:
+    """
+    Model class to store a single model information
+
+    Parameters
+    ----------
+    name: str
+        Name of the model
+    url: str
+        redirect url to the model on the official deepchem documentation
+    category: str
+        Category of the model
+    featurizers: list
+        List of featurizers supported by the model
+    backends: list
+        List of backends supported by the model
+    types: list
+        List of types supported by the model
+    model_id: int
+        Unique id of the model
+    """
+    name: str
+    url: str
+    category: str
+    featurizers: list
+    backends: list
+    types: list
+    model_id: int
+
+class ModelListEncoder(JSONEncoder):
+    """
+        ModelEncoder class to encode the Model object to JSON
+    """
+    def default(self, o):
+        """
+        Default function to encode the Model object to JSON
+
+        Returns
+        -------
+            json: str
+        """
+        return o.__dict__
+
+def convert_to_json(data, is_model=False):
+    """
+    Convert the list of data to JSON
+
+    Parameters
+    ----------
+        data: list
+        is_model: bool
+
+    Returns
+    -------
+        json: object
+    """
+    if is_model:
+        return json.dumps(data, indent=4, cls=ModelListEncoder)
+    data = list({item for sublist in data for item in sublist})
+    data = list(filter(None, data))
+    return json.dumps(data, indent=4)
+
+def scraper():
```

Comment:
We may not need scraper if we use the CSV file

Reply:
Renamed `scraper` to `fetch_data`. It is mainly required to fetch and format the data accordingly because of the inconsistencies of the csv file structures, to automate the process.